fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java

```diff
@@ -205,4 +214,48 @@ public static String convertPartitionFieldToPredicate(IcebergTable table, String
                     StatisticUtils.quoting(partitionColumn) + " < '" + endDateTimeStr + "'";
         }
     }
+
+    public static Expr toTransformExpr(IcebergTable table,
+                                       String partitionColumn,
+                                       SlotRef slotRef) {
+        PartitionField partitionField = table.getPartitionFiled(partitionColumn);
+        if (partitionField == null) {
+            throw new StarRocksConnectorException("Partition column %s not found in table %s.%s.%s",
+                    partitionColumn, table.getCatalogName(), table.getRemoteDbName(), table.getRemoteTableName());
+        }
+        IcebergPartitionTransform transform = IcebergPartitionTransform.fromString(partitionField.transform().toString());
+        if (transform == IcebergPartitionTransform.IDENTITY) {
+            return slotRef;
+        } else {
+            // transform is year, month, day, hour
+            Type partitiopnColumnType = table.getColumn(partitionColumn).getType();
+            Preconditions.checkState(partitiopnColumnType.isDateType(),
+                    "Partition column %s type must be date or datetime", partitionColumn);
+            String transformFuncName;
+            switch (transform) {
+                case HOUR:
+                    transformFuncName = FunctionSet.HOUR;
+                    break;
+                case DAY:
+                    transformFuncName = FunctionSet.DAY;
+                    break;
+                case MONTH:
+                    transformFuncName = FunctionSet.MONTH;
+                    break;
+                case YEAR:
+                    transformFuncName = FunctionSet.YEAR;
+                    break;
+                default:
+                    throw new StarRocksConnectorException("Unsupported partition transform to add: %s", transform);
```

Comment:
why we not support BUCKET/TRUNCATE ?

Reply:
I will do it in the next pr since this pr is too large.